### ISSUE TITLE
FeatureFinderMetaboIdent: allow quantifying charge 0 seeds (single traces)?

### ISSUE THREAD

**Issue by eeko-kon** [OpenMS/OpenMS#6049]
In the [MTBLS733](https://www.ebi.ac.uk/metabolights/MTBLS733/files) dataset where FFM can detect 784 features, FFMident only extracts 712 of them- many of the missing features were annotated with charge 0 from FFM and that charge is skipped by FFMident. 
FFM assigns zero charges when:
_"Mass traces that could not be assembled or low-intensity metabolites with only a monoisotopic mass trace to observe are left in the resulting @ref FeatureMap as singletons with the undefined charge state of 0."_

In the case of the dataset above, this is causing a lot of zero charges and so missing values. Does it make more sense to assign the value `0` than to assign those features with the charge `1`? 

By default, FFM should also ignore charges below 1 (see param `charge_lower_bound	1	Lowest charge state to consider` in [FFM](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_FeatureFinderMetabo.html) but I can still see a huge chunk of the features assigned as 0 charged). Or does this mean that it simply doesn't search for isotopes/reduce information for those features?

**Comment by jpfeuffer** [comment]
> In the case of the dataset above, this is causing a lot of zero charges and so missing values. Does it make more sense to assign the value 0 than to assign those features with the charge 1?

Would allowing FFMIdent to take charge 0 traces also work? I think this would make more sense than assigning a potentially wrong charge.

> By default, [FFM](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_FeatureFinderMetabo.html) should also ignore charges below 1 (see param charge_lower_bound 1 Lowest charge state to consider in FFM but I can still see a huge chunk of the features assigned as 0 charged). Or does this mean that it simply doesn't search for isotopes/reduce information for those features?

Maybe this needs a doc update. I assume charge 0 does not count as a charge since it stands for "unknown".

**Comment by eeko-kon** [comment]
> Would allowing FFMIdent to take charge 0 traces also work? I think this would make more sense than assigning a potentially wrong charge.

I just tweaked the Metabolite identification library (that is generated from the FFM files) and replaced all zero charges with 1  and managed to annotate 782 compounds after FFMident (784 after FFM).

**Comment by eeko-kon** [comment]
I am closing this issue since this is something that I ran into for the first time so frequently because of the very low intensities of the metabolites in the [MTBLS733](https://www.ebi.ac.uk/metabolights/MTBLS733/files) dataset.

**Comment by jpfeuffer** [comment]
Let's keep it open. I think allowing the quantification with a setting in FFMID is worthwhile.

**Comment by hendrikweisser** [comment]
FFMetId calculates the m/z value(s) for chromatogram extraction from mass and charge. That doesn't work when the charge is zero. (Which is okay, because molecules without a charge won't "fly" in a mass spec anyway.)
So you'd have to define what you want to happen for charge zero. Something like this might make sense: "If charge is zero, treat the target mass as the m/z value and extract one mass trace." (Normally we'd want at least two isotopic mass traces so their shapes and relative intensities can be compared, but that doesn't work for charge zero.)

**Comment by jpfeuffer** [comment]
True! Zero in this case means unknown due to one mass trace, yes. Thanks for clarifying. Would it work with just one mass trace (i.e. is the  comparison of the shapes optional)? Or do you think we would need more changes downstream?

**Comment by hendrikweisser** [comment]
> Would it work with just one mass trace (i.e. is the comparison of the shapes optional)?

Depends on what OpenSWATH does internally. In principle it should be possible, although some of the OpenSWATH scores wouldn't work. It could be tested after changing the minimum for the "extract:n_isotopes" parameter in FFMetId from 2 to 1.
